src/utils.ts

```diff
@@ -0,0 +1,118 @@
+import { breakpoint } from '@core-ds/primitives';
+import { css } from 'styled-components';
+import { Breakpoint, EventKeys, ResponsiveStyle, Style } from './types';
+
+const breakpointKeys: Array<keyof Breakpoint> = Object.keys(breakpoint) as any;
+
+export function isResponsiveStyle<Value>(x?: Style<Value>): x is ResponsiveStyle<Value> {
+   return x != null && typeof x !== 'string';
```

Comment:
```suggestion
   return typeof x === 'object';
```

Why not check for the type being an object? These should all be objects right?

Reply:
I'm afraid this won't work, as `typeof null` is also `'object'` (I know, javascript is weird)

Reply:
Would this work?
```suggestion
   return x && typeof x === 'object;
```

Reply:
You usually don't want to check for type `"object"`, as it is a pretty broad check (null, objects, arrays, classes, errors, dates, etc all are of type `object`).
If you want to check for an object to be a plain JSON like object (which is the expected value for a responsive style) the most accurate way that I know of is [`lodash.isPlainObject`](https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L12036). 
Having said that, I think that for the purpose of this algorithm `x && typeof x === 'object'` should work. Do you prefer it this way? Typescript will prevent wrong types anyway so I think either ways are fine